sklearn/census_online_prediction.py

```diff
@@ -0,0 +1,177 @@
+
+
+# [START categorical-feature-conversion]
+# Since the census data set has categorical features, we need to convert
+# them to numerical values. We'll use a list of pipelines to convert each
+# categorical column and then use FeatureUnion to combine them before calling
+# the RandomForestClassifier.
+categorical_pipelines = []
+
+# Each categorical column needs to be extracted individually and converted to a numerical value.
+# To do this, each categorical column will use a pipeline that extracts one feature column via
+# SelectKBest(k=1) and a LabelBinarizer() to convert the categorical value to a numerical one.
+# A scores array (created below) will select and extract the feature column. The scores array is
+# created by iterating over the COLUMNS and checking if it is a CATEGORICAL_COLUMN.
+for i, col in enumerate(COLUMNS[:-1]):
+    if col in CATEGORICAL_COLUMNS:
+        # Create a scores array to get the individual categorical column.
+        # Example:
+        #  data = [39, 'State-gov', 77516, 'Bachelors', 13, 'Never-married', 'Adm-clerical',
+        #         'Not-in-family', 'White', 'Male', 2174, 0, 40, 'United-States']
+        #  scores = [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+        #
+        # Returns: [['Sate-gov']]
+        scores = []
+        # Build the scores array
+        for j in range(len(COLUMNS[:-1])):
+            if i == j: # This column is the categorical column we want to extract.
+                scores.append(1) # Set to 1 to select this column
+            else: # Every other column should be ignored.
+                scores.append(0)
+        skb = SelectKBest(k=1)
+        skb.scores_ = scores
+        # Convert the categorical column to a numerical value
+        lbn = LabelBinarizer()
+        r = skb.transform(train_features)
+        lbn.fit(r)
+        # Create the pipeline to extract the categorical feature
+        categorical_pipelines.append(
+            ('categorical-{}'.format(i), Pipeline([
+                ('SKB-{}'.format(i), skb),
+                ('LBN-{}'.format(i), lbn)])))
+# [END categorical-feature-conversion]
+
+# [START create-pipeline]
+# Create pipeline to extract the numerical features
+skb = SelectKBest(k=6)
+# From COLUMNS use the features that are numerical
+skb.scores_ = [1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 1, 1, 0]
+categorical_pipelines.append(('numerical', skb))
+
+# Combine all the features using FeatureUnion
+preprocess = FeatureUnion(categorical_pipelines)
+
+# Create the classifier
+classifier = RandomForestClassifier()
+
+# Transform the features and fit them to the classifier
+classifier.fit(preprocess.transform(train_features), train_labels)
+
+# Create the overall model as a single pipeline
+pipeline = Pipeline([
+    ('union', preprocess),
+    ('classifier', classifier)
+])
+# [END create-pipeline]
+
+# [START export-model-locally]
+# Export the model to a file
+joblib.dump(pipeline, 'model.joblib')
+# [END export-model-locally]
+
```

Comment:
Everything below here should be sent to a 2nd file. 

If a user tries to run this, it will fail because the model is not yet on ML Engine.